go/scheduler/api/api.go

```diff
@@ -67,46 +67,53 @@ type CommitteeNode struct {
 type CommitteeKind uint8
 
 const (
-	// KindExecutor is an executor committee.
-	KindExecutor CommitteeKind = 0
+	// KindCompute is a compute committee.
+	KindInvalid CommitteeKind = 0
 
-	// KindStorage is a storage committee.
-	KindStorage CommitteeKind = 1
+	// KindComputeExecutor is an executor committee.
+	KindComputeExecutor CommitteeKind = 1
+
+	// KindComputeTxnScheduler is a transaction scheduler committee.
+	KindComputeTxnScheduler CommitteeKind = 2
 
-	// KindTransactionScheduler is a transaction scheduler committee.
-	KindTransactionScheduler CommitteeKind = 2
+	// KindComputeMerge is a merge committee.
+	KindComputeMerge CommitteeKind = 3
 
-	// KindMerge is a merge committee.
-	KindMerge CommitteeKind = 3
+	// KindStorage is a storage committee.
+	KindStorage CommitteeKind = 4
 
 	// MaxCommitteeKind is a dummy value used for iterating all committee kinds.
-	MaxCommitteeKind = 4
+	MaxCommitteeKind = 5
 )
 
 // NeedsLeader returns if committee kind needs leader role.
-func (k CommitteeKind) NeedsLeader() bool {
+func (k CommitteeKind) NeedsLeader() (bool, error) {
 	switch k {
-	case KindExecutor:
-		return false
-	case KindMerge:
-		return false
+	case KindInvalid:
+		return false, fmt.Errorf("scheduler: KindInvalid should not be queried for a leader")
```

Comment:
What about all the other kinds that are also invalid (e.g., greater than or equal to `MaxCommitteeKind`)?